jsk_perception/src/virtual_camera_mono.cpp

```diff
@@ -1,70 +1,34 @@
-#include <ros/ros.h>
-#include <dynamic_reconfigure/server.h>
-#include <jsk_topic_tools/log_utils.h>
+#include "jsk_perception/virtual_camera_mono.h"
+
 #include <jsk_topic_tools/rosparam_utils.h>
-#include <image_transport/image_transport.h>
-#include <image_geometry/pinhole_camera_model.h>
-#include <tf/transform_listener.h>
-#include <tf/transform_broadcaster.h>
 #include <cv_bridge/cv_bridge.h>
-#if ( CV_MAJOR_VERSION >= 4)
-#else
-#include <opencv/cv.hpp>
-#include <opencv/highgui.h>
-#endif
-
 #include <string>
 #include <vector>
-#include <boost/assign.hpp>
-#include <boost/thread.hpp>
-
-#include <geometry_msgs/Polygon.h>
-#include <geometry_msgs/PolygonStamped.h>
-
-#include <jsk_perception/VirtualCameraMonoConfig.h>
-
 
-class VirtualCameraMono
+namespace jsk_perception
 {
-  ros::NodeHandle nh_,private_nh_;
-  boost::shared_ptr<dynamic_reconfigure::Server<jsk_perception::VirtualCameraMonoConfig> > srv_;
-  image_transport::ImageTransport it_,it_priv_;
-  image_transport::CameraSubscriber sub_;
-  image_transport::CameraPublisher pub_;
-  int subscriber_count_;
-  tf::TransformListener tf_listener_;
-  image_geometry::PinholeCameraModel cam_model_;
-  tf::TransformBroadcaster tf_broadcaster_;
-  ros::Subscriber sub_trans_, sub_poly_;
-
-  tf::StampedTransform trans_; // transform to virtual camera
-  geometry_msgs::PolygonStamped poly_; // target polygon to transform image
-  int interpolation_method_;
-
-public:
-  VirtualCameraMono() : private_nh_("~"), it_(nh_), it_priv_(private_nh_), subscriber_count_(0)
+  void VirtualCameraMono::onInit()
   {
-    image_transport::SubscriberStatusCallback connect_cb = boost::bind(&VirtualCameraMono::connectCb, this, _1);
-    image_transport::SubscriberStatusCallback disconnect_cb = boost::bind(&VirtualCameraMono::disconnectCb, this, _1);
-    pub_ = it_priv_.advertiseCamera("image", 1, connect_cb, disconnect_cb);
+    DiagnosticNodelet::onInit();
+    pub_ = advertiseCamera(*pnh_, "image", 1, false);
```

Comment:
https://github.com/jsk-ros-pkg/jsk_recognition/runs/7291655426?check_suite_focus=true
Test failed.
Since jsk_common is released only up to 2.2.11 in kinetic etc., it is better to use `advertiseCamera(ros::NodeHandle& nh, const std::string& topic, int queue_size)`.
The `latch` option is `false` by default, so there is no problem.

Reply:
Thank you! Updated.